### DISCUSSION TITLE
Errors during inst installation, init and running

### DISCUSSION THREAD

**Discussion by anthonytietjen** [skiptools/skip#215] [General]
Using MacOS Ventura on Intel

**Getting a few errors during installation:**
% skip checkup
[✓] Skip version 1.0.8 (= 1.0.8)
[✓] macOS version 13.6.6 (> 13.5.0)
[!] Error running sysctl (Intel)
[✓] Swift version 5.9.2 (> 5.9.0)
[✓] Xcode version 15.1 (> 15.0.0)
[✓] Xcode tools SDKs: 8
[✓] Homebrew version 4.3.18 (> 4.1.0)
[✓] Gradle version 8.10 (> 8.6.0)
[!] Java version 11.0.16.1 (< 21.0.0) (check JAVA_HOME environment: /Library/Java/JavaVirtualMachines/_microsoft-11.jdk/Contents/Home)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2022.2
[✓] Android tools SDKs: 7
[✓] Resolve dependencies (43.12s)
[✓] Build hello-skip (44.1s)
[✓] Test Swift (23.95s)
[✗] Test Kotlin (93.13s)
[✓] Archive iOS ipa (47.79s)
[✓] Assemble HelloSkip-release.ipa 37 KB
[✓] Verify HelloSkip-release.ipa 37 KB
[✗] Assembling Android apk (84.32s)
[✗] Verify HelloSkip-release.apk: The file “app-release-unsigned.apk” couldn’t be opened because there is no such file.
[✓] Check Swift Package (1.07s)
[✓] Check Skip Updates: 1.0.8
[✗] Skip 1.0.8 checkup failed with 3 errors

**I thought maybe I could at least do iOS even if Android is failing, so if I try to proceed, then I get a couple more errors**
% skip init --appid=com.skiptest.skiptest skip-test SkipTest
Initializing Skip application skip-test
[✓] Resolve dependencies (29.31s)
[✗] Build skip-test (1.23s)
[✓] Check Swift Package (1.08s)
[✓] Created module SkipTest in /Users/anthony/repos/skip-test
[✗] Skip 1.0.8 init failed with 1 error

**Again, still hoping that iOS will work even if Android doesn't, but Xcode gives me this error when running:**
Missing package product 'SkipTestApp'

**Comment by marcprux** [comment]
> [!] Java version 11.0.16.1 (< 21.0.0) (check JAVA_HOME environment: /Library/Java/JavaVirtualMachines/_microsoft-11.jdk/Contents/Home)

This is likely the problem. Can you remove your JAVA_HOME from your environment (e.g., by commenting it out in the ~/.zshrc file) and starting a new Terminal, and trying `skip checkup` again?

**Comment by anthonytietjen** [comment]
Thanks for the idea. After removing JAVA_HOME from my environment and running skip checkup again in a new terminal window, I get one error (or is it a warning?) 

```
% skip checkup
[✓] Skip version 1.0.8 (= 1.0.8)
[✓] macOS version 13.6.6 (> 13.5.0)
[!] Error running sysctl (Intel)
[✓] Swift version 5.9.2 (> 5.9.0)
[✓] Xcode version 15.1 (> 15.0.0)
[✓] Xcode tools SDKs: 8
[✓] Homebrew version 4.3.18 (> 4.1.0)
[✓] Gradle version 8.10 (> 8.6.0)
[✓] Java version 22.0.2 (> 21.0.0)
[✓] Android Debug Bridge version 1.0.41 (> 1.0.40)
[✓] Android Studio version: 2022.2
[✓] Android tools SDKs: 7
[✓] Resolve dependencies (24.36s)
[✓] Build hello-skip (37.15s)
[✓] Test Swift (21.92s)
[✓] Test Kotlin (202.88s)
[✓] Archive iOS ipa (67.51s)
[✓] Assemble HelloSkip-release.ipa 37 KB
[✓] Verify HelloSkip-release.ipa 37 KB
[✓] Assembling Android apk (321.49s)
[✓] Verify HelloSkip-release.apk 13.2 MB
[✓] Check Swift Package (1.43s)
[✓] Check Skip Updates: 1.0.8
[!] Skip 1.0.8 checkup with 1 warning
```

The third line above says Error, but the final line says warning, so hoping for the best I tried to run skip init again and I got two errors:

```
% skip init --appid=com.skiptest.skiptest skip-test SkipTest
Initializing Skip application skip-test
[✓] Resolve dependencies (26.21s)
[✗] Build skip-test (1.13s)
[✓] Check Swift Package (0.99s)
[✓] Created module SkipTest in /Users/anthony/repos/swift/skip-test
[✗] Skip 1.0.8 init failed with 1 error
```

I wasn't sure these errors were blockers or not, so I opened Xcode and tried to run it and got the same error as earlier:

```
Missing package product 'SkipTestApp'
```

**Comment by marcprux** [comment]
I think the problem is that one of our built-in frameworks is `skip.test`, which your project module name is conflicting with. When I try that command and look at the error output, I see:

> error: multiple targets named 'SkipTest' in: 'skip', 'skip-test'

So I think you just need to pick a different name. E.g.:

```
skip init --appid=com.skipdemo.skipdemo skip-demo SkipDemo
```

**Comment by anthonytietjen** [comment]
@marcprux using a different name has solved the issue. I was able to run the hello world boilerplate on both iOS and Android. Thank you!

**Comment by wcwynn** [comment]
Was the 3rd line an error, or can safely ignore? what does it mean?
```
% skip checkup
[✓] Skip version 1.0.8 (= 1.0.8)
[✓] macOS version 13.6.6 (> 13.5.0)
[!] Error running sysctl (Intel)
```

**Reply by marcprux** [reply]
Yes, the `sysctl` warning can be ignored. That is known to happen on Intel machines.
